pytorch_lightning/trainer/trainer.py

```diff
@@ -1427,7 +1432,8 @@ def _call_teardown_hook(self) -> None:
         if self.datamodule is not None:
             self.datamodule.teardown(stage=fn)
 
-        self.call_hook("teardown", stage=fn)
+        self._call_hook(self, "teardown", stage=fn)
```

Comment:
passing `self` like this is very roundabout. it's already an instance method. `self` is accessible without needing to pass this arg.

Reply:
yeah I agree. this can be remedied by creating three diff methods as discussed above